k2/python/csrc/torch/v2/autograd/arc_sort.h

```diff
@@ -0,0 +1,81 @@
+/**
+ * @brief A wrapper around k2::ArcSort to support autograd
+ *
+ * @copyright
+ * Copyright      2021  Xiaomi Corp.  (authors: Fangjun Kuang)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef K2_PYTHON_CSRC_TORCH_V2_AUTOGRAD_ARC_SORT_H
+#define K2_PYTHON_CSRC_TORCH_V2_AUTOGRAD_ARC_SORT_H
+
+#include "k2/csrc/fsa_algo.h"
+#include "k2/csrc/ragged_ops.h"
+#include "k2/csrc/tensor.h"
+#include "k2/csrc/tensor_ops.h"
+#include "k2/python/csrc/torch/torch_util.h"
+#include "k2/python/csrc/torch/v2/ragged_arc.h"
+
+using namespace torch::autograd;
+
+namespace k2 {
+
+// see https://pytorch.org/tutorials/advanced/cpp_autograd
+class ArcSortFunction : public torch::autograd::Function<ArcSortFunction> {
```

Comment:
This is how we do `backprop` for `ArcSort`.

Reply:
Ah OK, cool.